src/Tests/dotnet-MsiInstallation.Tests/WorkloadSetTests.cs

```diff
@@ -294,6 +293,52 @@ public void InstallWithVersionAndSkipManifestUpdate()
                 .And.HaveStdErrContaining("--sdk-version");
         }
 
+        [Fact]
+        public void InstallWithVersionWhenPinned()
```

Comment:
This is the perfect test for testing this change. Simple, elegant, and it works. It'd be nice to make it run in CI, but this should be fine.